### DISCUSSION TITLE
Logging/Emitting Events from a worker

### DISCUSSION THREAD

**Discussion by dyllan-to-you** [jasnell/piscina#127] [Q&A] [answered]
Hi! I'm currently using piscina to run a queue of long-running tasks, and I would like to be able to get logs out from the worker about a task as the task runs.

The way I see it, my options are:
- routing stdout and stderr from the worker to my main process and handling the output from there
- emitting events from the worker and handling the event in my main process

The second option is a preferable, largely because my workers have a dependency on a C++ library that decides it likes to spam stdout while it runs. I'm not entirely sure how I'd go about doing this. Any ideas/suggestions? Is this something piscina supports or will I have to do some other weird hack

**Comment by jasnell** [comment, marked as answer]
Take a look at the progress example in the examples directory :-) It largely illustrates how to accomplish what you're looking to do.

There's not really anything we can do about the C++ writing out to stdio as that is bypassing the Node.js stdio re-routing used for worker threads.

**Reply by dyllan-to-you** [reply]
The MessageChannel works perfectly! Thanks for the awesome documentation & great examples!

OffTopic, but are you aware of any projects or integrations that provide a worker-like interface + queue, but support having tasks be run from some sort of distributed cluster?
